115-update-KafkaMirrorMaker2-resource-to-better-correspond-its-use.md

```diff
@@ -0,0 +1,212 @@
+    * The `alias`, `bootstrapServers`, `tls`, `authentication` and `config` sections will remain unchanged.
+* `.spec.mirrors[].source` section for configuring the source cluster.
+  The source cluster configuration itself will remain identical to the configuration we use today in `.spec.clusters`.
+
+The `.spec.target`, `.spec.mirrors[].source` and `.spec.mirrors` fields will be required in the `v1` CRD API.
+Within the `.spec.target` section, the `alias`, `bootstrapServers`, `groupId`, `configStorageTopic`, `statusStorageTopic`, and `offsetStorageTopic` will be required in both `v1beta2` and `v1` (this whole section will be optional in `v1beta2`, so this will not break backwards compatibility).
+To allow early migration to the new API while the `v1beta2` API is still in use, the `spec.connectCluster` field will be changed to not be required anymore in `v1beta2` (it is completely removed in `v1`).
+
+The following example shows the new `KafkaMirrorMaker2` layout:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker-2
+spec:
+  version: 4.0.0
+  replicas: 1
+  target:
+    alias: east
+    bootstrapServers: east-kafka-bootstrap:9092
+    groupId: my-mm2-group
+    configStorageTopic: my-mm2-config
+    statusStorageTopic: my-mm2-status
+    offsetStorageTopic: my-mm2-offsets
+    config:
+      config.storage.replication.factor: -1
+      offset.storage.replication.factor: -1
+      status.storage.replication.factor: -1
+  mirrors:
+    - source:
+        alias: "west"
+        bootstrapServers: west-kafka-bootstrap:9092
+      sourceConnector:
+        tasksMax: 1
+        config:
+          replication.factor: -1
+          offset-syncs.topic.replication.factor: -1
+          sync.topic.acls.enabled: "false"
+          refresh.topics.interval.seconds: 600
+      checkpointConnector:
+        tasksMax: 1
+        config:
+          checkpoints.topic.replication.factor: -1
+          sync.group.offsets.enabled: "false"
+          refresh.groups.interval.seconds: 600
+      topicsPattern: ".*"
+      groupsPattern: ".*"
+    - source:
+        alias: "south-pole"
+        bootstrapServers: south-pole-kafka-bootstrap:9092
+      sourceConnector:
+        tasksMax: 1
+        config:
+          replication.factor: -1
+          offset-syncs.topic.replication.factor: -1
+          sync.topic.acls.enabled: "false"
+          refresh.topics.interval.seconds: 600
+      checkpointConnector:
+        tasksMax: 1
+        config:
+          checkpoints.topic.replication.factor: -1
+          sync.group.offsets.enabled: "false"
+          refresh.groups.interval.seconds: 600
+      topicsPattern: ".*"
+      groupsPattern: ".*"
+```
+
+The new fields will be immediately available to Strimzi users.
+When both the new and old fields are configured, Strimzi will always prefer the configuration from the new fields.
+During this phase, we would expect the custom resource to use either the new or the old API.
```

Comment:
_"During this phase, the custom resource must use either the new or the old API, not both."_ ?